x/incentives/keeper/distribute_test.go

```diff
@@ -1131,3 +1132,223 @@ func (s *KeeperTestSuite) IncentivizeInternalGauge(poolIds []uint64, epochDurati
+
+	// create 3 internal Gauge
+	var internalGauges []uint64
+	for i := 0; i <= 2; i++ {
+		internalGauge := s.CreateNoLockExternalGauges(clPool.GetId(), sdk.NewCoins(), s.TestAccs[1], uint64(1)) // gauge id = 2,3,4
+		internalGauges = append(internalGauges, internalGauge)
+	}
+
+	// create group gauge
+	groupGauge, err := s.App.IncentivesKeeper.CreateGroupGauge(s.Ctx, sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(100_000_000))), uint64(1), s.TestAccs[1], internalGauges) // gauge id = 3
+	s.Require().NoError(err)
+
+	// Call AllocateAcrossGauges
+	// Allocate 100 osmo across 3 internal gauges evenly
+	s.App.IncentivesKeeper.AllocateAcrossGauges(s.Ctx)
+
+	groupGaugePostAllocate, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, groupGauge)
+	s.Require().NoError(err)
+
+	s.Require().Equal(groupGaugePostAllocate.Coins.Sub(groupGaugePostAllocate.DistributedCoins), sdk.Coins(nil))
+
+	for _, gauge := range internalGauges {
+		internalGauge, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, gauge)
+		s.Require().NoError(err)
+
+		s.Require().Equal(internalGauge.Coins.Sub(internalGauge.DistributedCoins), expectedInternalGaugeCoins)
+	}
+
+}
+
+/*
+Test Structure
+Step 1:
+Create 3 internal-gauges that'll we'll add to split incentives
+Create 1 regular no-lock perp gauge to make sure it's flow is not affected
+
+Step 2:
+Create Group Gauge with 100_000_000uosmo paid over 3 epoch, equally split between 3 internal gauges.
+It also links internal-gauge with groupGauge Will look something like this: {groupGaugeId, [internalGaugeId1, internalGaugeId2, internalGaugeId3]}
+
+Step 3:
+Let epoch 1 pass we expect:
+
+	internalGauge1 to recieve 11_111_111uosmo, internalGauge2 to recieve 11_111_111uosmo, internalGauge3 to recieve 11_111_111uosmo
+	decrement GroupGauge coins from 100_000_000 - 33_333_333 = 66_666_667
+	We also expect, regular gauge incentive to be distributed.
+	Hence, total 3 incentiveRecord for CL pools (2 from internal-gauges, 1 from regular gauge) & 1 GAMM lock incentive.
+
+Step 4:
+Let epoch 2 pass, we expect:
+
+	internalGauge1 to recieve 11_111_111uosmo, internalGauge2 to recieve 11_111_111uosmo, internalGauge3 to recieve 11_111_111uosmo
+	incentiveGauge1, incentiveGauge2, incentiveGauge3 should be empty
+	decrement GroupGauge coins from 66_666_667 - 33_333_333 = 33_333_334
+	Hence, total 5 incentiveRecord for CL pools
+
+Step 5:
+Let epoch 3 pass, we expect:
+
+	internalGauge1 to recieve 11_111_111uosmo, internalGauge2 to recieve 11_111_111uosmo, internalGauge3 to recieve 11_111_111uosmo
+	incentiveGauge1, incentiveGauge2, incentiveGauge3 should be empty
+	decrement GroupGauge coins from 33_333_334 - 33_333_333 = 1
+	Hence, total 7 incentiveRecord for CL pools
+*/
+func (s *KeeperTestSuite) TestCreateGroupGaugeAndDistribute() {
+	var (
+		epochInfo = s.App.IncentivesKeeper.GetEpochInfo(s.Ctx)
+
+		expectedCoinsDistributed       = sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(11_111_111)))
```

Comment:
In general, having new logic rely entirely on a single hard coded test vector is probably not a sufficient testing practice. Happy to discuss what might be a more robust approach offline if you'd like

Reply:
Refactored to make it more robust and readable: https://github.com/osmosis-labs/osmosis/pull/6085/files#diff-d5967ddef40ae47ef4140b9c47026cac85b85d9d04173b271c6c23dd182abdbcR1200